token/program-2022/src/processor.rs

```diff
@@ -1465,188 +1466,206 @@ impl Processor {
+                }
+                TokenInstruction::UiAmountToAmount { ui_amount } => {
+                    msg!("Instruction: UiAmountToAmount");
+                    Self::process_ui_amount_to_amount(accounts, ui_amount)
+                }
+                TokenInstruction::Reallocate { extension_types } => {
+                    msg!("Instruction: Reallocate");
+                    reallocate::process_reallocate(program_id, accounts, extension_types)
+                }
+                TokenInstruction::MemoTransferExtension => {
+                    memo_transfer::processor::process_instruction(program_id, accounts, &input[1..])
+                }
+                TokenInstruction::CreateNativeMint => {
+                    msg!("Instruction: CreateNativeMint");
+                    Self::process_create_native_mint(accounts)
+                }
+                TokenInstruction::InitializeNonTransferableMint => {
+                    msg!("Instruction: InitializeNonTransferableMint");
+                    Self::process_initialize_non_transferable_mint(accounts)
+                }
+                TokenInstruction::InterestBearingMintExtension => {
+                    interest_bearing_mint::processor::process_instruction(
+                        program_id,
+                        accounts,
+                        &input[1..],
+                    )
+                }
+                TokenInstruction::CpiGuardExtension => {
+                    cpi_guard::processor::process_instruction(program_id, accounts, &input[1..])
+                }
+                TokenInstruction::InitializePermanentDelegate { delegate } => {
+                    msg!("Instruction: InitializePermanentDelegate");
+                    Self::process_initialize_permanent_delegate(accounts, delegate)
+                }
+                TokenInstruction::TransferHookExtension => {
+                    transfer_hook::processor::process_instruction(program_id, accounts, &input[1..])
+                }
+                TokenInstruction::ConfidentialTransferFeeExtension => {
+                    confidential_transfer_fee::processor::process_instruction(
+                        program_id,
+                        accounts,
+                        &input[1..],
+                    )
+                }
+                TokenInstruction::WithdrawExcessLamports => {
+                    msg!("Instruction: WithdrawExcessLamports");
+                    Self::process_withdraw_excess_lamports(program_id, accounts)
+                }
+                TokenInstruction::MetadataPointerExtension => {
+                    metadata_pointer::processor::process_instruction(
+                        program_id,
+                        accounts,
+                        &input[1..],
+                    )
+                }
             }
+        } else if let Ok(instruction) = TokenMetadataInstruction::unpack(input) {
```

Comment:
Sick! This line here is literally keying on the interface's instruction set (imported). So clean 🧼